obsreport/obsreport_receiver.go

```diff
@@ -163,23 +175,42 @@ func (rec *Receiver) endOp(
 	span := trace.SpanFromContext(receiverCtx)
 
 	if rec.level != configtelemetry.LevelNone {
-		var acceptedMeasure, refusedMeasure *stats.Int64Measure
-		switch dataType {
-		case config.TracesDataType:
-			acceptedMeasure = obsmetrics.ReceiverAcceptedSpans
-			refusedMeasure = obsmetrics.ReceiverRefusedSpans
-		case config.MetricsDataType:
-			acceptedMeasure = obsmetrics.ReceiverAcceptedMetricPoints
-			refusedMeasure = obsmetrics.ReceiverRefusedMetricPoints
-		case config.LogsDataType:
-			acceptedMeasure = obsmetrics.ReceiverAcceptedLogRecords
-			refusedMeasure = obsmetrics.ReceiverRefusedLogRecords
+		if rec.otelEnabled {
+			var acceptedMeasure, refusedMeasure syncint64.Counter
+			switch dataType {
+			case config.TracesDataType:
+				acceptedMeasure = obsmetrics.OtelReceiverAcceptedSpans
+				refusedMeasure = obsmetrics.OtelReceiverRefusedSpans
+			case config.MetricsDataType:
+				acceptedMeasure = obsmetrics.OtelReceiverAcceptedMetricPoints
+				refusedMeasure = obsmetrics.OtelReceiverRefusedMetricPoints
+			case config.LogsDataType:
+				acceptedMeasure = obsmetrics.OtelReceiverAcceptedLogRecords
+				refusedMeasure = obsmetrics.OtelReceiverRefusedLogRecords
```

Comment:
extract separate funcs "recordWithOtel" and "recordWithOC" to help with readability.

Reply:
done.